pkg/supervisor/supervisor_nix.go

```diff
@@ -0,0 +1,98 @@
+//go:build !windows
```

Comment:
I think it is not necessary to have explicit build tags and the OS specific suffix in the filename. So probably having files supervisor_posix and supervisor_windows COULD work here :)

Btw, why do we have two `go:build` and `build:`? 
We can safely use only the one which works for the latest go version since we don't provide our code as a library.
Otherwise we have three different build constraint sources: filename, new style tag, old style tag and we need to keep them more or less in sync.

What do you think, if we try to have only one build tag style or even check, may be with a correct naming (_posix and _windows) it would not require tags at all?

Reply:
> Btw, why do we have two go:build and build:?
Quite frankly, I just grepped and copied it from other places :) 

> I think it is not necessary to have explicit build tags and the OS specific suffix in the filename. So probably having files supervisor_posix and supervisor_windows COULD work here :)

Turns out _windows.go is respected by default but _posix.go / _unix.go is not :/ . Although _linux.go is honored.
I added the //build:  https://go-review.googlesource.com/c/go/+/389935 because I fear calling it _linux.go will break macOS and I don't really want to add yet another dummy file  for darwin...